test/parallel_api/numeric/numeric.ops/in_place_scan.pass.cpp

```diff
@@ -19,64 +19,479 @@
 
 #include "support/test_config.h"
 #include "support/utils.h"
+#include "support/scan_serial_impl.h"
 
 #if TEST_DPCPP_BACKEND_PRESENT
-#    include "support/utils_sycl.h"
-#    include "support/sycl_alloc_utils.h"
-#    include "support/scan_serial_impl.h"
+#include <CL/sycl.hpp>
+#endif
+using namespace TestUtils;
+
+// TODO: replace data generation with random data and update check to compare result to
+// the result of a serial implementation of the algorithm
+template <typename Iterator1, typename Size>
+void
+initialize_data(Iterator1 host_keys, Size n)
+{
+    const Size kStartVal = 1;
+    for (Size i = 0; i != n; ++i)
+        host_keys[i] = kStartVal + i;
+}
+
+DEFINE_TEST_1(test_inclusive_scan_non_inplace, BinaryOperation)
```

Comment:
Is it possible to keep just one test definition  with adding parameters "inclusive/exclusive" and "inplace/outplace"? I guess it is pretty simple.

Reply:
From my point of view it's not good solution because each of tests will more complicated.
For example, we should make dependence of error check messages from this test param and so on.